### ISSUE TITLE
mpt2sas not working now but it did before

### ISSUE THREAD

**Issue by dperez-sct** [pocopico/rp-ext#53]
**TL;DR mpt2sas doesn't work for me and before it did. I have only removed vmxnet3 extension.**

Weeks ago I got DSM 7 up and running with the mpt2sas and vmxnet3 extension. A few days ago I upgraded from esxi to proxmox and was trying to change the network card from vmxnet3 to virtio to take advantage of gigabit speed.
With my previous configuration I have removed the vmxnet3 extension and added the virtio extension. I haven't been able to get DSM to boot, getting kernel panics on a few occasions. After many tests, it seems that there is "something" that has changed in the mpt2sas extension that generating an image now causes the DSM to fail me.

Thanks for the help and the hard work!

Data:

**OLD/WorkingDSM7:**

`root@Nassie:~$ dmesg | grep mpt
[    1.167448] <redpill/uart_swapper.c:425> Disabling preempt & locking console
[    1.308729] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=07 dev=00 fn=00
[    1.338428] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=08 dev=00 fn=00
[    1.365640] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=09 dev=00 fn=00
[    1.393048] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=0a dev=00 fn=00
[    2.538492] mpt2sas version 14.100.00.00 loaded
[    2.568869] mpt2sas0: 64 BIT PCI BUS DMA ADDRESSING SUPPORTED, total mem (4022040 kB)
[    2.570542] mpt2sas 0000:00:10.0: irq 45 for MSI/MSI-X
[    2.591369] mpt2sas0-msix0: PCI-MSI-X enabled: IRQ 45
[    2.592341] mpt2sas0: iomem(0x00000000feb10000), mapped(0xffffc90000120000), size(65536)
[    2.593704] mpt2sas0: ioport(0x000000000000e000), size(256)
[    2.710901] mpt2sas0: Allocated physical memory: size(6173 kB)
[    2.711974] mpt2sas0: Current Controller Queue Depth(2748), Max Controller Queue Depth(2879)
[    2.713401] mpt2sas0: Scatter Gather Elements per IO(128)
[    2.773601] mpt2sas0: LSISAS2008: FWVersion(07.15.08.00), ChipRevision(0x03), BiosVersion(00.00.00.00)
[    2.775264] mpt2sas0: Dell 6Gbps SAS HBA: Vendor(0x1000), Device(0x0072), SSVID(0x1028), SSDID(0x1F1C)
[    2.776867] mpt2sas0: Protocol=(Initiator,Target), Capabilities=(Raid,TLR,EEDP,Snapshot Buffer,Diag Trace Buffer,Task Set Full,NCQ)
[    2.779358] mpt2sas0: sending port enable !!
[    4.321942] mpt2sas0: host_add: handle(0x0001), sas_addr(0x5000000080000000), phys(8)
[   10.447073] mpt2sas0: port enable: SUCCESS`

root@Nassie:~$ lsmod | grep mpt
mpt2sas               137614  13 

**NEW/NON WorkingDSM7:**

`DiskStation> dmesg | grep mpt
[    1.144279] <redpill/uart_swapper.c:425> Disabling preempt & locking console
[    1.286056] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=07 dev=00 fn=00
[    1.316668] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=08 dev=00 fn=00
[    1.345533] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=09 dev=00 fn=00
[    1.373387] <redpill/virtual_pci.c:420> Attempting to add vPCI device [printed below] @ bus=0a dev=00 fn=00
[    3.758320] mpt2sas version 20.103.01.00 loaded
[    3.789665] mpt2sas_cm0: 64 BIT PCI BUS DMA ADDRESSING SUPPORTED, total mem (4022040 kB)
[    3.847645] mpt2sas_cm0: MSI-X vectors supported: 1, no of cores: 4, max_msix_vectors: -1
[    3.848985] genirq: Flags mismatch irq 0. 00000080 (mpt2sas0-msix0) vs. 00015a20 (timer)
[    3.850304] mpt2sas0-msix0: unable to allocate interrupt 0!
[    3.851410] mpt2sas0: IO-APIC enabled: IRQ 11
[    3.852117] mpt2sas_cm0: iomem(0x00000000feb10000), mapped(0xffffc90000140000), size(65536)
[    3.853422] mpt2sas_cm0: ioport(0x000000000000e000), size(256)
[    3.949734] mpt2sas_cm0: Allocated physical memory: size(6336 kB)
[    3.950759] mpt2sas_cm0: Current Controller Queue Depth(2808),Max Controller Queue Depth(2879)
[    3.952121] mpt2sas_cm0: Scatter Gather Elements per IO(128)
[    4.132455] mpt2sas_cm0: log_info(0x30030100): originator(IOP), code(0x03), sub_code(0x0100)
[    4.133411] mpt2sas_cm0: log_info(0x30030100): originator(IOP), code(0x03), sub_code(0x0100)
[    4.133853] mpt2sas_cm0: LSISAS2008: FWVersion(07.15.08.00), ChipRevision(0x03), BiosVersion(00.00.00.00)
[    4.133855] mpt2sas_cm0: Protocol=(
[    4.149161] mpt2sas_cm0: sending port enable !!
[   19.155117] mpt2sas_cm0: _config_request: timeout
[   19.160619] mpt2sas_cm0: sending diag reset !!
[   20.490478] mpt2sas_cm0: diag reset: SUCCESS
[   50.596166] mpt2sas_cm0: _base_event_notification: timeout
[   50.601862] mpt2sas_cm0: failure at /opt/dsm-kernel/mpt3sas-16.100.01.00/mpt3sas_scsih.c:5062/_scsih_sas_host_add()!
[  304.152106] mpt2sas_cm0: port enable: FAILED with timeout (timeout=300s)

DiskStation> lsmod
Module                  Size  Used by    Tainted: PF 
usbhid                 26503  0 
hid                    79391  1 usbhid
bromolow_synobios      70388  0 
adt7475                30106  0 
i2c_i801               11053  0 
nfsv3                  23713  0 
nfs                   130128  1 nfsv3
lockd                  72839  2 nfsv3,nfs
sunrpc                202956  3 nfsv3,nfs,lockd
i40e                  350790  0 
ixgbe                 274157  0 
igb                   180080  0 
i2c_algo_bit            5168  0 
e1000e                212512  0 
dca                     4576  2 ixgbe,igb
vxlan                  17560  0 
ip_tunnel              11368  1 vxlan
vfat                   10287  0 
fat                    51522  1 vfat
sg                     25890  0 
etxhci_hcd            135146  0 
mpt2sas               196102  0 
usb_storage            48966  0 
xhci_hcd               85621  0 
uhci_hcd               23950  0 
ehci_pci                3904  0 
ehci_hcd               47493  1 ehci_pci
usbcore               208589  7 usbhid,etxhci_hcd,usb_storage,xhci_hcd,uhci_hcd,ehci_pci,ehci_hcd
usb_common              1560  1 usbcore
redpill               146375  0`

**Comment by dperez-sct** [comment]
More info, with this checkout i'm already generated a new working image. With this my problem has solved for now:

https://github.com/pocopico/rp-ext/commit/89f89901b75a898bac14effb2de75a2929afc403

I've downloaded the repo and put ir on a local http server and added to toolchain and this worked for me, with this old commit.

Regards!
